### DISCUSSION TITLE
Infinite log output to file

### DISCUSSION THREAD

**Discussion by s2terminal** [zauberzeug/nicegui#1717] [Q&A] [answered]
### Question

Hi. Thank you for the great library.
When I use logging to output logs to a file, I get an infinite number of logs that detect file changes. Is there a misconfiguration?

```python
import logging
from nicegui import ui

logging.basicConfig(filename="dev.log", level=logging.DEBUG)

ui.label("hello")

ui.run()
```

Once a file is edited, the following logs are output indefinitely.

```
DEBUG:watchfiles.main:1 change detected: {(<Change.modified: 2>, '/app/dev.log')}
DEBUG:watchfiles.main:1 change detected: {(<Change.modified: 2>, '/app/dev.log')}
DEBUG:watchfiles.main:1 change detected: {(<Change.modified: 2>, '/app/dev.log')}
```

**Comment by rodja** [comment]
Either you save the logging file into another directory or you use the `uvicorn_reload_excludes` parameter in `ui.run`: https://nicegui.io/documentation/run#:~:text=uvicorn_reload_excludes

**Reply by s2terminal** [reply]
I have set uvicorn_reload_excludes but the logging does not stop. Am I setting it up wrong?

```python
ui.run(uvicorn_reload_excludes="dev.log")
```

**Reply by s2terminal** [reply]
If it is a uvicorn issue, I will investigate that. Thank you.

**Reply by falkoschindler** [reply]
@s2terminal Please make sure to restart the app manually after setting `uvicorn_reload_excludes`, because auto-reload won't apply changes to this parameter.

**Reply by s2terminal** [reply]
I am using Docker and building a container and restarting the app did not improve the situation.

**Comment by natankeddem** [comment, marked as answer]
related issue https://github.com/zauberzeug/nicegui/issues/1533

**Reply by falkoschindler** [reply]
Ah, I see. The log file _is_ excluded from uvicorn reloads. But since
```py
logging.basicConfig(filename="dev.log", level=logging.DEBUG)
```
seems to set the logging level globally, the detected file change is logged anyway, causing a change detection which is logged again and so on.

@s2terminal Can you try @natankeddem's workaround from https://github.com/zauberzeug/nicegui/issues/1533#issuecomment-1701846652, i.e. not using the global logger? Does it help?

**Reply by s2terminal** [reply]
I see. I used `logging.getLogger(name)` and used the logger object. Problem solved. Thank you very much.

**Comment by rodja** [comment]
With `uvicorn_reload_excludes` you can only prevent the app from reloading. That you log expands infinitely and would also happen in any app started with uvicorn and autoreload (as explained in https://github.com/zauberzeug/nicegui/discussions/1717#discussioncomment-7190989). So this is not NiceGUI specific.
